content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Limit Order Line Quantity
+description: How-To Guide to limit order line quantity in Vendr, the eCommerce solution for Umbraco v8+
+---
+
+In this guide we will be looking at Validation events in Vendr to be able to limit order line quantity based on the existing stock value on the product and the existing quantities of the product in the cart.
+
+## ProductAddValidationHandler
+
+When adding a product to cart we need to verify the product is in stock and the customer not already has the remaining quantities in cart.
+
+````csharp
+public class ProductAddValidationHandler : ValidationEventHandlerBase<ValidateOrderProductAdd>
+{
+    private readonly IUmbracoContextAccessor _context;
+
+    public ProductAddValidationHandler(IUmbracoContextAccessor context)
+    {
+        _context = context;
+    }
+
+    public override void Validate(ValidateOrderProductAdd evt)
+    {
+        var order = evt.Order;
+        var productReference = evt.ProductReference;
+
+        var product = _context.UmbracoContext.Content.GetById(new Guid(productReference));
+        var stock = product.Value<decimal?>("stock");
```

Comment:
should probably not bother using the property value here, I'd recommend using the `IProductService` service instead, which would need injecting into the handler, then call

     _productService.GetProductStock(productReference);

Reply:
I haven't tested that, but I guess it should be working in a similar way.

Reply:
Fixed in https://github.com/vendrhub/vendr-documentation/pull/32/commits/828a28319531d4507abfa77ce18fdba19c8ac808